src/components/orator/js/Orator.js

```diff
@@ -0,0 +1,571 @@
+            }
+        },
+        components: {
+            tts: {
+                type: "fluid.textToSpeech",
+                options: {
+                    invokers: {
+                        queueSpeech: {
+                            funcName: "fluid.orator.domReader.queueSpeech",
+                            args: [
+                                "{that}",
+                                "fluid.textToSpeech.queueSpeech",
+                                "{arguments}.0",
+                                "{arguments}.1",
+                                "{arguments}.2"
+                            ]
+                        }
+                    }
+                }
+            }
+        },
+        invokers: {
+            readFromDOM: {
+                funcName: "fluid.orator.domReader.readFromDOM",
+                args: ["{that}", "{that}.container"]
+            },
+            removeHighlight: {
+                funcName: "fluid.orator.domReader.unWrap",
+                args: ["{that}.dom.highlight"]
+            },
+            resetParseQueue: {
+                funcName: "fluid.orator.domReader.resetParseQueue",
+                args: ["{that}"]
+            },
+            highlight: {
+                funcName: "fluid.orator.domReader.highlight",
+                args: ["{that}", "{arguments}.0"]
+            },
+            play: {
+                funcName: "fluid.orator.domReader.speak",
+                args: ["{that}"]
+            },
+            pause: {
+                funcName: "fluid.orator.domReader.pause",
+                args: ["{that}"]
+            }
+        },
+        listeners: {
+            "{tts}.events.utteranceOnEnd": [{
+                listener: "{that}.resetParseQueue",
+                namespace: "resetParseQueue"
+            }, {
+                listener: "{that}.removeHighlight",
+                namespace: "removeHighlight"
+            }],
+            "{tts}.events.utteranceOnBoundary": {
+                listener: "{that}.highlight",
+                args: ["{arguments}.0.charIndex"],
+                namespace: "highlightUtteranceText"
+            },
+            "onDestroy.detachRange": {
+                "this": "{that}.range",
+                method: "detach"
+            }
+        }
+    });
+
+    fluid.orator.domReader.speak = function (that) {
+        if (that.model.paused) {
+            that.tts.resume();
+        } else if (!that.model.speaking) {
+            that.readFromDOM();
+        }
+    };
+
+    fluid.orator.domReader.pause = function (that) {
+        if (that.model.speaking && !that.model.paused) {
+            that.tts.pause();
+        }
+    };
+
+    // Accepts a speechFn (either a function or function name), which will be used to perform the
+    // underlying queuing of the speech. This allows the SpeechSynthesis to be replaced (e.g. for testing)
+    fluid.orator.domReader.queueSpeech = function (that, speechFn, text, interrupt, options) {
```

Comment:
No public docs for this function, including the interesting "interrupt" argument. Also its argument list has become very bulky. Better to structure this entirely as a (that, options) structure with named arguments.
Also, it would be better to structure this activity as an event pipeline with namespaced listeners, which will make it easier to chop and change implementations for mocking. In that way we should be able to get rid of the peculiar "speechFn" argument

Reply:
Talked about the event pipeline in the channel.
https://botbot.me/freenode/fluid-work/2018-07-05/?msg=101795899&page=1